active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+        {
+            PropertyChanged?.Invoke(
+                    this, new PropertyChangedEventArgs(propertyName));
+            Validate(propertyName);
+        }
+    }
+    #endregion
+
+    #region Populate errors (C# - DataAnnotations)
+    private Task Validate(string propertyName)
+    {
+        ClearErrors();
+
+
+        return Task.Run(() =>
+        {
+            var val = GetPropertyValue(propertyName);
+            var results = new ObservableCollection<ValidationResult>();
+            var context = new ValidationContext(this, null, null) 
+                            { MemberName = propertyName };
+    
+            // Validator is from the .Net System.ComponentModel.DataAnnotations namespace
+            Validator.TryValidateProperty(val, context, results);
+
+            if (results.Any())
+            {
+                    foreach (var result in results)
+                    {
+                        Errors[propertyName].Add(new ValidationError(result.ErrorMessage));
+                    }
+            }
+
+                OnErrorsChanged(propertyName);
+            }
+
+        }
+    }
+    #endregion
+
+    #region privates
+    private object GetPropertyValue([CallerMemberName] string propertyName = "")
+    {
+        return _values.ContainsKey(propertyName) ? _values[propertyName] : null;
+    }
+
+    #endregion
+    }
+}
+```
+
+## Custom validation UI
+
+
+#### XAML
+A developer wants to create their own visuals for the error message style and not use the control’s visuals at all. They may want to do this to place messages far from the control or something else.
+
+To do this, the developer needs to turn off the system error visuals and then display the content from the errors collection in their own UI.
+
+![Custom UI](customuiexample.png)
+
+```xaml
+<TextBox 
+        x:Name="Name" 
+        Header="Name: " 
+        PlaceholderText="Enter Name..." 
+
+        Text="{x:Bind   ViewModel.Person.Name, 
+                        UpdateSourceTrigger=PropertyChanged, 
+                        Mode=TwoWay}" 
+         UseSystemValidationVisuals="False"
+        />
+
+<!-- Custom error message template -->
+<ItemsControl
+    ItemsSource="{x:Bind Name.(Validation.Errors), Mode=OneWay}"
+    Foreground="Red">
+
+    <ItemsControl.ItemTemplate>
+        <DataTemplate x:DataType="ValidationError">
+            <StackPanel Orientation="Horizontal" Spacing="5">
+                <Border Background="Red" Margin="3" CornerRadius="10" 
+                        Width="12" Height="12" VerticalAlignment="Top">
+                    <TextBlock Text="!" Foreground="White" 
+                               HorizontalAlignment="Center" VerticalAlignment="Center"/>
+                </Border>
+                <TextBlock Text="{x:Bind ErrorMessage, Mode=OneWay}"
+                           />
+            </StackPanel>
+        </DataTemplate>
+    </ItemsControl.ItemTemplate>
+</ItemsControl>
+```
+
+## Customizing the error template
```

Comment:
How does this work with different `InputValidationKind` options?

Reply:
The input validation kind determine if the template is displayed inline or in tooltip.

Reply:
So I'm clear that I understand correctly, you're saying that if this is specified it becomes the template that's used when in the error state, regardless of the `InputValidationKind`?

What then about needing to allow for coping with variations in the UI for different focus states when also in the error state?

Reply:
Is there a need for a control primitive which can be styled for this Validation message?  It could also be used on other controls that are not TextBox?